modules/lsp/src/main/scala/playground/lsp/ModelLoader.scala

```diff
@@ -39,15 +36,20 @@ object ModelLoader {
       dependencies = buildConfig.smithyPlayground.foldMap(_.extensions),
       repositories =
         buildConfig.mavenRepositories ++ buildConfig.maven.foldMap(_.repositories).map(_.url),
-    )
+    ),
+    isolated = false,
   )
 
   private def makeClassLoaderForJars(
-    jars: List[File]
+    jars: List[File],
+    isolated: Boolean,
   ): URLClassLoader =
     new URLClassLoader(
       jars.map(_.toURI().toURL()).toArray,
-      getClass().getClassLoader(),
+      if (isolated)
+        null
```

Comment:
@baccata do you think this is a decent way to avoid loading upstream models when loading a user's dependencies?

I had issues when using `ModelDiscovery` manually (see removed code block below). I'll try to reproduce it somewhere later, but in general it seemed like all the url-uri-path-file-string conversions were causing trouble when your jar was in a path like `/Users/kubukoz/Library/Caches/Coursier/v1/https/aws%40xxxxxxxxx.d.codeartifact.us-east-1/...`.

Reply:
also see https://github.com/disneystreaming/smithy4s/pull/846, it seems to be fine in smithy4s too